File: CS/CustomizeExportAction.Module/Controllers/CustomizeExportController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;

using DevExpress.ExpressApp.SystemModule;

using DevExpress.XtraPrinting;

namespace CustomizeExportAction.Module.Controllers {
   public partial class CustomizeExportController : ViewController {
      public CustomizeExportController() {
         InitializeComponent();
         RegisterActions(components);
         TargetViewType = ViewType.ListView;
      }
      private ExportController exportController;
      protected override void OnActivated() {
         base.OnActivated();
         exportController = Frame.GetController<ExportController>();
         exportController.CustomGetDefaultFileName += exportController_CustomGetDefaultFileName;
         exportController.CustomExport += new EventHandler<CustomExportEventArgs>(CustomExport);
    }

      protected virtual void CustomExport(object sender, CustomExportEventArgs e) {
         //Customize Export Options
         if (e.ExportTarget == ExportTarget.Xls) {
            XlsExportOptions options = (XlsExportOptions)e.ExportOptions;
            if (options == null) {
               options = new XlsExportOptions();
            }
            options.SheetName = View.Caption;
            options.ShowGridLines = true;
            e.ExportOptions = options;
         }
      }
      void exportController_CustomGetDefaultFileName(object sender, CustomGetDefaultFileNameEventArgs e) {
#if EASYTEST
			 //Provide a custom file name
         e.FileName = e.FileName + "_06.25.12";
#else
          //Provide a custom file name
         e.FileName = e.FileName + "_" + DateTime.Now.ToString("MM.dd.yy");
#endif
      }
      protected override void OnDeactivated() {
         exportController.CustomGetDefaultFileName -= exportController_CustomGetDefaultFileName;
         exportController.CustomExport -= new EventHandler<CustomExportEventArgs>(CustomExport);
         base.OnDeactivated();
      }
   }
}

File: VB/CustomizeExportAction.Module/Controllers/CustomizeExportController.vb
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base

Imports DevExpress.ExpressApp.SystemModule

Imports DevExpress.XtraPrinting

Namespace CustomizeExportAction.Module.Controllers
   Partial Public Class CustomizeExportController
       Inherits ViewController

      Public Sub New()
         InitializeComponent()
         RegisterActions(components)
         TargetViewType = ViewType.ListView
      End Sub
      Private exportController As ExportController
      Protected Overrides Sub OnActivated()
         MyBase.OnActivated()
         exportController = Frame.GetController(Of ExportController)()
         AddHandler exportController.CustomGetDefaultFileName, AddressOf exportController_CustomGetDefaultFileName
         AddHandler exportController.CustomExport, AddressOf CustomExport
      End Sub

      Protected Overridable Sub CustomExport(ByVal sender As Object, ByVal e As CustomExportEventArgs)
         'Customize Export Options
         If e.ExportTarget = ExportTarget.Xls Then
            Dim options As XlsExportOptions = CType(e.ExportOptions, XlsExportOptions)
            If options Is Nothing Then
               options = New XlsExportOptions()
            End If
            options.SheetName = View.Caption
            options.ShowGridLines = True
            e.ExportOptions = options
         End If
      End Sub
      Private Sub exportController_CustomGetDefaultFileName(ByVal sender As Object, ByVal e As CustomGetDefaultFileNameEventArgs)
#If EASYTEST Then
             'Provide a custom file name
         e.FileName = e.FileName & "_06.25.12"
#Else
          'Provide a custom file name
         e.FileName = e.FileName & "_" & Date.Now.ToString("MM.dd.yy")
#End If
      End Sub
      Protected Overrides Sub OnDeactivated()
         RemoveHandler exportController.CustomGetDefaultFileName, AddressOf exportController_CustomGetDefaultFileName
         RemoveHandler exportController.CustomExport, AddressOf CustomExport
         MyBase.OnDeactivated()
      End Sub
   End Class
End Namespace
